packages/edit-site/src/components/resizable-frame/index.js

```diff
@@ -33,108 +33,124 @@ const HANDLE_STYLES_OVERRIDE = {
 };
 
 const MIN_FRAME_SIZE = 340;
-const lerp = ( a, b, amount ) => {
-	return a + ( b - a ) * amount;
-};
 
-function ResizableFrame( { isFull, children } ) {
+function calculateNewHeight( width, initialAspectRatio ) {
+	const lerp = ( a, b, amount ) => {
+		return a + ( b - a ) * amount;
+	};
+
+	// Calculate the intermediate aspect ratio based on the current width.
+	const lerpFactor =
+		1 -
+		Math.max(
+			0,
+			Math.min(
+				1,
+				( width - MIN_FRAME_SIZE ) / ( 1300 - MIN_FRAME_SIZE )
+			)
+		);
+
+	// Calculate the height based on the intermediate aspect ratio
+	// ensuring the frame arrives at a 9:19.5 aspect ratio.
+	const intermediateAspectRatio = lerp(
+		initialAspectRatio,
+		9 / 19.5,
+		lerpFactor
+	);
+
+	return width / intermediateAspectRatio;
+}
+
+function ResizableFrame( { isFullWidth, children } ) {
 	const [ frameSize, setFrameSize ] = useState( {
 		width: '100%',
 		height: '100%',
 	} );
+	const [ startingWidth, setStartingWidth ] = useState();
```

Comment:
We can probably add some inline comments to make clear the distinction between `startingWidth` and `initialComputedWidthRef`

```suggestion
	// The width of the resizable element when a new resize interaction start
	const [ startingWidth, setStartingWidth ] = useState();
```